src/game/board/boardSituation.js

```diff
@@ -1,14 +1,31 @@
-export default class BoardSituation {
-    constructor() {
+export default class BoardSituation {
+        constructor(firstBoardPosition) {
+            if(newBoardPosition instanceof  BoardPosition){
+                this.boardPosition=firstBoardPosition;
+                this.boardSituation=[];
+            }
+            else {
+                 window.alert('Incorrect data given');
+            }
+        }
+        get position() {
+            // TODO: this function should return current position on the board as an 8x8 array of ChessPiece objects
+            return boardPosition;
+        }
```

Comment:
Ponieważ (przynajmniej na razie) wszędzie poza Twoimi klasami pozycja jest tablicą, to tutaj powinnaś zwracać tablicę (przy obacnych nazwach zmiennych byłoby to `return boardPosition.position` ).